server/app/com/xsn/explorer/migrations/MigrationRunner.scala

```diff
@@ -81,15 +82,21 @@ class BlockRewardsMigration @Inject()(
   private def storeBlockReward(rpcBlock: rpc.Block[_]) = {
     val result = for {
       extractionMethod <- blockService.extractionMethod(rpcBlock).toFutureOr
-      reward <- blockService.getBlockRewards(rpcBlock, extractionMethod).toFutureOr
-      _ <- db.insertBlockRewards(rpcBlock.hash, reward).toFutureOr
+      reward <- blockService
+        .getBlockRewards(rpcBlock, extractionMethod)
+        .map {
+          case Good(reward) => Good(Some(reward))
+          case Bad(_) => Good(None)
+        }
+        .toFutureOr
+      _ <- Future.successful(Good(reward.foreach(r => db.insertBlockRewards(rpcBlock.hash, r)))).toFutureOr
```

Comment:
There is no need to do all, this is actually enough.
```
_ = reward.foreach(r => db.insertBlockRewards(rpcBlock.hash, r))
```

Reply:
but that way if `db.insertBlockRewards(rpcBlock.hash, r)` fails for some reason we would lose that error wont we?